### PR TITLE
UssGetAcl, UssExtAttr and UssSetAcl implementations

### PR DESCRIPTION
**Pull request by kostrja**

### CODE REVIEW

File: src/main/java/zowe/client/sdk/zosfiles/uss/methods/UssGetAcl.java [hunk 1]
```diff
@@ -9,5 +9,94 @@
...
+
+            final String url = "https://" + connection.getHost() + ":" + connection.getZosmfPort() +
+                    ZosFilesConstants.RESOURCE + ZosFilesConstants.RES_USS_FILES +
+                    EncodeUtils.encodeURIComponent(FileUtils.validatePath(fileNamePath));
+
+            final Map<String, Object> getFaclMap = new HashMap<>();
+            params.getType().ifPresent(type -> getFaclMap.put("type", type));
+            params.getUser().ifPresent(user -> getFaclMap.put("user", user));
+            if (params.getUseCommas())
+                getFaclMap.put("useCommas", params.getUseCommas());
```
Comment (frankgiordano) [inline src/main/java/zowe/client/sdk/zosfiles/uss/methods/UssGetAcl.java:88]: "useCommas" is incorrect. The IBM web documentation says it should be "use-commas"

You need to have code that parses the Json responsephrase output with either "\n" or ","

File: src/main/java/zowe/client/sdk/zosfiles/uss/methods/UssGetAcl.java [hunk 2]
```diff
@@ -9,5 +9,94 @@
...
+            if (params.getSuppressHeader())
+                getFaclMap.put("suppressHeader", params.getSuppressHeader());
```
Comment (frankgiordano) [inline src/main/java/zowe/client/sdk/zosfiles/uss/methods/UssGetAcl.java:90]: incorrect. key value is "suppress-header" not "suppressHeader"

File: src/main/java/zowe/client/sdk/zosfiles/uss/methods/UssGetAcl.java [hunk 3]
```diff
@@ -9,5 +9,94 @@
...
+            if (params.getSuppressBaseAcl())
+                getFaclMap.put("suppressBaseAcl", params.getSuppressBaseAcl());
```
Comment (frankgiordano) [inline src/main/java/zowe/client/sdk/zosfiles/uss/methods/UssGetAcl.java:92]: incorrect key value, it should be "suppress-baseacl" see IBM web documentation please

File: src/main/java/zowe/client/sdk/zosfiles/uss/methods/UssGetAcl.java [hunk 4]
```diff
@@ -9,5 +9,94 @@
...
+
+        /**
+        * Get the ACL for a USS file or directory
+        *
+        * @param fileNamePath file name with path
+         * @param params GetAclParams object to drive the request
+        * @return Response object
+        * @throws Exception processing error
+        */
+        public Response getAcl(String fileNamePath, GetAclParams params) throws Exception {
```
Comment (frankgiordano) [inline src/main/java/zowe/client/sdk/zosfiles/uss/methods/UssGetAcl.java:75]: make this method getAclCommon name and add parent caller methods with returning responsephrase as a parsed string output

File: src/main/java/zowe/client/sdk/zosfiles/uss/methods/UssGetAcl.java [hunk 5]
```diff
@@ -9,5 +9,94 @@
...
+            ValidateUtils.checkNullParameter(fileNamePath == null, "path is null");
+            ValidateUtils.checkIllegalParameter(fileNamePath.trim().isBlank(), "path is empty");
+            ValidateUtils.checkNullParameter(params == null, "params is null");
+
+            final String url = "https://" + connection.getHost() + ":" + connection.getZosmfPort() +
+                    ZosFilesConstants.RESOURCE + ZosFilesConstants.RES_USS_FILES +
+                    EncodeUtils.encodeURIComponent(FileUtils.validatePath(fileNamePath));
+
+            final Map<String, Object> getFaclMap = new HashMap<>();
```
Comment (frankgiordano) [inline src/main/java/zowe/client/sdk/zosfiles/uss/methods/UssGetAcl.java:84]: add map entry for request type

File: src/main/java/zowe/client/sdk/zosfiles/uss/input/GetAclParams.java [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+    public String toString() {
+        return "GetAclParams{" +
+                "type=" + type +
+                ", user=" + user +
+                ", useCommas=" + useCommas +
+                ", suppressHeader=" + suppressHeader +
+                ", suppressBaseAcl=" + suppressBaseAcl +
+                '}';
+    }
+
```
Comment (frankgiordano) [inline src/main/java/zowe/client/sdk/zosfiles/uss/input/GetAclParams.java:92]: add javadoc for builder object too.. see other builder object I added javadocs throughout now..

**Comment by frankgiordano** [comment]
AI assistant might made the map key words for you which were incorrect I am assuming lol

File: src/main/java/zowe/client/sdk/zosfiles/uss/methods/UssExtAttr.java [hunk 7]
```diff
@@ -9,5 +9,87 @@
...
+
+    /**
+     * Extends the attributes of a file or directory
+     *
+     * @param set   one of more of the following: alps
+     * @param reset one or more of the following: alps
+     */
+    public Response extendAttributes(String targetPath, String set, String reset) throws Exception {
+        ValidateUtils.checkNullParameter(targetPath == null, "targetPath is null");
+        ValidateUtils.checkIllegalParameter(targetPath.trim().isEmpty(), "targetPath not specified");
```
Comment (frankgiordano) [inline src/main/java/zowe/client/sdk/zosfiles/uss/methods/UssExtAttr.java:74]: we are no longer using trim() anymore. project wide it uses isBlank() method call instead.. please replace

File: src/main/java/zowe/client/sdk/zosfiles/uss/methods/UssExtAttr.java [hunk 8]
```diff
@@ -9,5 +9,87 @@
...
+
+        final String url = "https://" + connection.getHost() + ":" + connection.getZosmfPort()
+                + ZosFilesConstants.RESOURCE + ZosFilesConstants.RES_USS_FILES + FileUtils.validatePath(targetPath);
+
+        final Map<String, String> jsonMap = new HashMap<>();
+        jsonMap.put("request", "extattr");
+        if (set != null && !set.trim().isEmpty())
+            jsonMap.put("set", set);
+        if (reset != null && !reset.trim().isEmpty())
+            jsonMap.put("reset", reset);
```
Comment (frankgiordano) [inline src/main/java/zowe/client/sdk/zosfiles/uss/methods/UssExtAttr.java:84]: this doesnt look pretty the if checks.. I suggest rework this method to execute either a set or reset action, require one to be specified. and create a new method call display(String targetPath) to execute the request without a set and reset specified which would just return a string output of the display attributes as noted in the IBM web documentation.

File: src/main/java/zowe/client/sdk/zosfiles/uss/methods/UssExtAttr.java [hunk 9]
```diff
@@ -9,5 +9,87 @@
...
+        this.request = request;
+    }
+
+    /**
+     * Extends the attributes of a file or directory
+     *
+     * @param set   one of more of the following: alps
+     * @param reset one or more of the following: alps
+     */
+    public Response extendAttributes(String targetPath, String set, String reset) throws Exception {
```
[the remark this answers is not in the document]
Comment (frankgiordano) [reply src/main/java/zowe/client/sdk/zosfiles/uss/methods/UssExtAttr.java:72]: commenting further from another comment below lets force setandr reset to be specified via two methods: 
 public Response set(String targetPath, String value) throws Exception {
 public Response reset(String targetPath, String value) throws Exception {

Overall this class should contain three verb action methods:

 public Response display(String targetPath) throws Exception {
 public Response set(String targetPath, String value) throws Exception {
 public Response reset(String targetPath, String value) throws Exception {

and a private common method called executeRequest that performs the request code:

        if (request == null) {
            request = ZosmfRequestFactory.buildRequest(connection, ZosmfRequestType.PUT_JSON);
        }
        request.setUrl(url);
        request.setBody(new JSONObject(jsonMap).toString());

and returns the Response object. 

This is a much better design of the API calls for this method than having the end user use null or empty value to drive the request from one method call as you initially put in place.

File: src/main/java/zowe/client/sdk/zosfiles/uss/methods/UssGetAcl.java [hunk 10]
```diff
@@ -9,5 +9,127 @@
...
+            ValidateUtils.checkConnection(connection);
+            ValidateUtils.checkNullParameter(request == null, "request is null");
+            this.connection = connection;
+            if (!(request instanceof PutJsonZosmfRequest)) {
+                throw new Exception("PUT_JSON request type required");
+            }
+            this.request = request;
+        }
+
+        public String getAclWithCommas(String fileNamePath) throws Exception {
```
Comment (frankgiordano) [inline src/main/java/zowe/client/sdk/zosfiles/uss/methods/UssGetAcl.java:71]: At this time I think this method's existent and naming is awkward. Let's remove it and add a boolean value to the main getAcl method second parameter to specify to receive the output with commas or not and handle the logic there.

File: src/main/java/zowe/client/sdk/zosfiles/uss/methods/UssExtAttr.java (hunk 9 above)
[the remark this answers is not in the document]
Comment (kostrja) [reply src/main/java/zowe/client/sdk/zosfiles/uss/methods/UssExtAttr.java:72]: Done except display, what should this method do?

**Review by frankgiordano — APPROVED** [review]

**Pull request merged**